### PR TITLE
remove documentation about setuptools integration Refs #5534 Fixes #3814

### LINKED ISSUE #3814: recommended pytest-runner in setup_requires means packages fail to install often
The recommendation to add `pytest-runner` to `setup_requires` means that all users of that package end up with an unnecessary pytest-runner package installed. This is bad because it bypasses pip hashes and [`--trusted-host`](https://github.com/pypa/pip/issues/4156)

https://docs.pytest.org/en/latest/goodpractices.html#integrating-with-setuptools-python-setup-py-test-pytest-runner

https://github.com/MechanicalSoup/MechanicalSoup/pull/224
https://github.com/rxcomm/pyaxo/issues/26
https://github.com/jpadilla/pyjwt/issues/179

### PR DESCRIPTION
**Pull request by graingert**
integrating with `python setup.py test` is not good practice!

- [x] Target the `master` branch for bug fixes, documentation updates and trivial changes.
- [ ] Target the `features` branch for new features and removals/deprecations.
- [ ] Include documentation when adding new features.
- [ ] Include new tests or update existing tests when applicable.

Unless your change is trivial or a small documentation fix (e.g.,  a typo or reword of a small section) please:

- [ ] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.
- [ ] Add yourself to `AUTHORS` in alphabetical order;

### CODE REVIEW

**Comment by graingert** [comment]
https://github.com/pytest-dev/pytest/issues/5534

File: doc/en/goodpractices.rst [hunk 1]
```diff
@@ -219,101 +219,4 @@ against your source code checkout, helping to detect packaging
 glitches.
 
 
-Integrating with setuptools / ``python setup.py test`` / ``pytest-runner``
```
Comment (nicoddemus) [inline doc/en/goodpractices.rst]: I would keep this section, perhaps renaming it to: 
```rst
Integrating with setuptools using ``pytest-runner``
```

But we should still remove the `Manual Integration` section.

Comment (graingert) [reply doc/en/goodpractices.rst]: From @jaraco 

> Yes, I'm in favor of removing the section altogether. I don't see any value in keeping a reference to pytest-runner.

Comment (nicoddemus) [reply doc/en/goodpractices.rst]: Oh OK then, I missed that comment. 👍 

In that case let's remove it altogether.

**Comment by nicoddemus** [comment]
Thanks!

**Pull request merged**